packages/react-components/src/hooks/useAlarms/transformers/mapAlarmRuleExpression.ts

```diff
@@ -0,0 +1,45 @@
+import { ComparisonOperator } from '@aws-sdk/client-iot-events';
+import { AlarmData } from '../types';
+
+type RuleExpression<
+  InputPropertyName extends string = string,
+  Operator extends string = string,
+  ThresholdValue extends number = number
+> = `${InputPropertyName} ${Operator} ${ThresholdValue}`;
+
+const COMPARISON_SYMBOL = {
+  [ComparisonOperator.EQUAL]: '=',
+  [ComparisonOperator.LESS]: '<',
+  [ComparisonOperator.LESS_OR_EQUAL]: '<=',
+  [ComparisonOperator.GREATER]: '>',
+  [ComparisonOperator.GREATER_OR_EQUAL]: '>=',
+  [ComparisonOperator.NOT_EQUAL]: '!=',
+};
+
+const getComparisonSymbol = (operator?: string) => {
+  if (!operator) return undefined;
+
+  const operatorSymbol = COMPARISON_SYMBOL[operator as ComparisonOperator];
+
+  if (operatorSymbol == null) return undefined;
+
+  return operatorSymbol;
+};
+
+export const mapAlarmRuleExpression = ({
+  inputProperty,
+  thresholds,
+  models,
+}: AlarmData): RuleExpression | undefined => {
+  const inputPropertyName = inputProperty?.at(0)?.property.name;
+  const threshold = thresholds?.at(-1)?.value?.doubleValue;
+  const operator = getComparisonSymbol(
+    models?.at(-1)?.alarmRule?.simpleRule?.comparisonOperator
+  );
+
+  if (inputPropertyName != null && operator != null && threshold != null) {
+    return `${inputPropertyName} ${operator} ${threshold}`;
+  }
+
+  return undefined;
```

Comment:
Returning nothing would be functionally the same as returning undefined though. I wanted to make it obvious in code that the value will be undefined if those 3 variables did not have a value, in the case someone was trying to debug why it's not showing up.

Reply:
Why don't we always return some string, and just replace the field with '_' if it's not defined?

The table could have:
```
_ > 30
windSpeed _ 30
windSpeed > _
```

We would display some error if something happened, or that signals to the user that there is some misconfigured alarm.

Reply:
And if all 3 aren't defined then why not return an empty string?

Reply:
By default table handles replacing an undefined value with a `'-'`, which is why I wanted to make it explicit that we should return undefined if we can't represent the `ruleExpression`. I don't think we should show a partial expression though, because it won't provide enough information for someone to debug the status.